presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/GroupTrait.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.sql.planner.iterative;
+
+import static java.util.Objects.requireNonNull;
+
+public interface GroupTrait
+{
+    enum Type
```

Comment:
instead of `enum Type` I think we could have some kind of:
```
class TraitType<T extends Trait>
``` 

this way we can implement `getTrait` method with type safety:
```
public <T extends Trait> T getTrait(TraitType<T> traitType)
```